python/ray/util/client/server/proxier.py

```diff
@@ -560,61 +570,120 @@ def modify_connection_info_resp(self,
         return modified_resp
 
     def Datapath(self, request_iterator, context):
+        cleanup_requested = False
```

Comment:
This function is too way big. Could we split it to make it better organized?

Reply:
Sounds good, I can handle in a follow up PR, but let me know if you want it in this one

Reply:
Probably we can just split in this PR if you don't mind which can improve the readability